### PR TITLE
Fixed hid in sysmodules

### PR DESCRIPTION
**Pull request by jakibaki**

### CODE REVIEW

**Comment by yellows8** [comment]
Sysmodules are supposed to disable hid init/exit. https://github.com/switchbrew/libnx/blob/master/nx/source/runtime/init.c#L89

**Comment by jakibaki** [comment]
Without this pr hidInitialize would fail if called from a sysmodule so the sysmodule couldn't read the (for example) joycon inputs.

This fixes that and allows the joycon inputs to be read as a sysmodule.

File: nx/source/services/hid.c [hunk 1]
```diff
@@ -49,8 +49,9 @@ Result hidInitialize(void)
     u64 AppletResourceUserId;
 
     rc = appletGetAppletResourceUserId(&AppletResourceUserId);
+    // If this failed (for example because we're a sysmodule) AppletResourceUserId stays zero
     if (R_FAILED(rc))
-        return rc;
+        AppletResourceUserId = 0;
```
Comment (yellows8) [inline nx/source/services/hid.c]: Does hidSetNpadJoyAssignmentModeDual() work with this same change if the below check is removed?

Comment (jakibaki) [reply nx/source/services/hid.c]: Yeah, it does!

I'll update my pr with better changes.

File: nx/source/services/hid.c [hunk 2]
```diff
@@ -569,8 +570,7 @@ Result hidMergeSingleJoyAsDualJoy(HidControllerID id0, HidControllerID id1) {
 
     rc = appletGetAppletResourceUserId(&AppletResourceUserId);
     if (R_FAILED(rc))
-        return rc;
-
+        AppletResourceUserId = 0;
```
Comment (fincs) [inline nx/source/services/hid.c]: The newline was mistakenly removed here.

Comment (jakibaki) [reply nx/source/services/hid.c]: you're right, I'll fix that.

**Pull request merged**